rfcs/0000-switch-postgresql-from-ec2-to-rds.md

```diff
@@ -0,0 +1,190 @@
+
+## Proposed Solution
+[Proposed Solution]: #proposed-solution
+
+[Amazon's Relational Database Service (RDS)](https://aws.amazon.com/rds/) offerings are basically,
+  "click a button and get a PostgreSQL server" -- they're managed services wherein Amazon runs and mostly manages RDS for you.
+RDS is offered in several different "flavours": PostgreSQL, MySQL, their "Aurora" product, and others.
+These services remove many, though not all, of the hassles of running a database reliably: they manage the OS, patching, and backups.
+That leaves us still on the hook to manage: major upgrades, maintenance processes like vaccuuming, and tuning.
+For a team as small as ours, this counts as a rather significant win:
+  we're freeing our time and energy up to focus on more important things.
+
+One of the most important considerations here is cost: RDS isn't free.
+For details, please see the attached
+  [RFC postgresql-on-rds Addendum: AWS Pricing](https://docs.google.com/spreadsheets/d/1eeMVqy9xdctcJO80M1JGVqd6apUaza-hfKGtIKLhIdE/edit?usp=sharing)
+  spreadsheet, but at a high-level:
+
+* RDS Aurora charges about twice as much for compute.
+* RDS Aurora does not charge for extra storage for each compute instance; all instances within a region share the same storage.
+    * As storage-per-instance is currently project to be our largest and fastest-growing AWS expense,
+      this would provide us with very large savings.
+* RDS Aurora also introduces a new charge of $0.20 per 1M I/O operations (both read & write).
+  With our projected number of DB queries, this will not be a major cost driver.
+    * It could _become_ a major cost driver if our usage wildly exceeds projections.
+      But, in that scenario, having the more reliable managed DB-as-a-service backing us becomes an even bigger concern that justifies the cost.
+
+Overall, RDS Aurora will likely save us anywhere from 27% to 71% on our AWS costs, through 2020-06: it's a major win on cost control.
+
+It's worth mentioning that we're actually already leveraging RDS for several things:
+
+* RDS has powered the "old sandbox" all along.
+* RDS was also used as the database for the initial load development and benchmarking that was done
+
+This change is an "under the hood" thing only; it does not directly impact external users of our systems.
+
+### Proposed Solution: Detailed Design
+[Proposed Solution: Detailed Design]: #proposed-solution-detailed-design
+
+For the most important pieces of our systems, the Data Pipeline and Data Server,
+  RDS Aurora is a fully-compatible switch: all they need to change are their JDBC URLs.
+For our infrastructure automation, RDS will be a major win, as we'll go from having to orchestrate an EC2 instance,
+  operating system, _and_ PostgreSQL configuration to just asking AWS to, "give us an RDS instance, please."
+
+Once ready, The deployment/transition to RDS Aurora should be simple and smooth:
+
```

Comment:
Suggest that we include a trial phase with Aurora. Setup an instance and see how it performs. This might already be the intent, but it could be made more clear in this proposal.